backend/src/graphql/resolvers/UserResolver.ts

```diff
@@ -84,6 +87,18 @@ export class UserResolver {
     if (!result.success) {
       throw new Error(result.data)
     }
+    const qluser = new User(result.data.user)
+    const user = new DbUser()
+    user.pubkey = Buffer.from(fromHex(qluser.pubkey))
+    user.email = qluser.email
+    user.firstName = qluser.firstName
+    user.lastName = qluser.lastName
+    user.username = qluser.username
+
+    const repository = getRepository(DbUser)
+    repository.save(user).catch(() => {
+      throw new Error('error saving user')
+    })
```

Comment:
What about the case that the user already exists? Does this never happen?

Also I suggest the following shorthand which should work:

```suggestion
    user.save().catch(() => {
      throw new Error('error saving user')
    })
```

Reply:
Ja diese Version läuft solange wir das Active Record Pattern benutzen
"Using the Active Record approach, you define all your query methods inside the model itself, and you save, remove, and load objects using model methods. "
Aber mir scheint das du und moritz sich darüber einig seit, das ihr lieber das DataMapper Pattern benutzen wollt 
"Using the Data Mapper approach, you define all your query methods in separate classes called "repositories", and you save, remove, and load objects using repositories."
und daher dachte kann ich ja neue Sachen gleich mit dem DataMapper Pattern definieren, damit wir das später nicht wieder umbauen müssen. 

Oder wollen wir das vermischen?

Reply:
In this case no.
Because if the user already exist

> s? Does this never happen?



> What about the case that the user already exists? Does this never happen?
In this case not. Because if the user already exist login-server will be return an error and the function will exit before calling this.